finality-provider/service/fp_instance.go

```diff
@@ -316,9 +373,17 @@ func (fp *FinalityProviderInstance) hasVotingPower(b *types.BlockInfo) (bool, er
 			zap.Uint64("block_height", b.Height),
 		)
 
+		if fp.GetStatus() == proto.FinalityProviderStatus_ACTIVE {
+			fp.MustSetStatus(proto.FinalityProviderStatus_INACTIVE)
```

Comment:
Nit: not a huge fan of this, it's a bit hidden away that this function updates state. I suggest making it return the voting power or something like (bool, uint64, err), and then the function caller can decide what to do with it.

Alternatively, add comment saying that it also updates local status

Reply:
Good point. Did some refactoring in [757156e](https://github.com/babylonlabs-io/finality-provider/pull/233/commits/757156e076588a39a50b42476ef87261648070d0)